manageiq-operator/deploy/operator.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: manageiq-operator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: manageiq-operator
+  template:
+    metadata:
+      labels:
+        name: manageiq-operator
+    spec:
+      serviceAccountName: manageiq-operator
+      containers:
+        - name: manageiq-operator
+          # Replace this with the built image name
+          image: docker.io/chainzeeli/manageiq-operator:v0.0.1
+          command:
+          - manageiq-operator
+          imagePullPolicy: Always
+          env:
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
```

Comment:
If this is always going to be the current namespace, can we use the namespace file in the pod instead of passing additional env vars?

Reply:
which pod are you referring to?

Reply:
Every pod. ref (https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#directly-accessing-the-rest-api-1)
> the default namespace to be used for namespaced API operations is placed in a file at /var/run/secrets/kubernetes.io/serviceaccount/namespace in each container

Reply:
We use this for kube API operations in manageiq https://github.com/ManageIQ/manageiq/blob/1283535e41099a9c34cebdb27aac83c151fbb70d/lib/container_orchestrator/object_definition.rb#L88-L91

Reply:
In this case should I just make a container startup script to set the ENVs?

Reply:
> We use this for kube API operations in manageiq https://github.com/ManageIQ/manageiq/blob/1283535e41099a9c34cebdb27aac83c151fbb70d/lib/container_orchestrator/object_definition.rb#L88-L91

In this case should I just make a container startup script to set the ENVs?

Reply:
Oh is a `WATCH_NAMESPACE` env var required for the operator? I figured you would just go read the file rather than using an env var at all.

Reply:
> Oh is a `WATCH_NAMESPACE` env var required for the operator? I figured you would just go read the file rather than using an env var at all.

yeah all the yaml files in this directory are generated by the SDK, so the ENVs are parts of the architecture

Reply:
Okay, this is fine for now then.

Reply:
wait actually we can't do everything inside `Dockerfile`.  when building the binary from go source code, we need some system environment variables, which we need to resolve outside of Dockerfile, so we either need a wrapper script for `docker build` or pass all the args in one `docker build` command.